### PR TITLE
Fixed deprecation warnings on webpack@5.

### PR DESCRIPTION
**Pull request by sanex3339**
Fixed https://github.com/TypeStrong/ts-loader/issues/1194

### CODE REVIEW

**Comment by sanex3339** [comment]
Don't sure how to test it, especially because previous checks for webpack@5 were without tests.

File: src/index.ts [hunk 1]
```diff
@@ -644,11 +645,19 @@ function getTranspilationEmit(
...
+      { module },
       loaderContext.context
     );
 
-    loaderContext._module.errors.push(...errors);
+    /**
+     * Since webpack 5, the `errors` property is deprecated,
+     * so we can check if some methods for reporting errors exist.
+     */
+    if (!!module.addError) {
```
Comment (johnnyreilly) [inline src/index.ts:656]: Do we need the `!!` here?

```suggestion
    if (module.addError) {
```

File: src/instances.ts [hunk 2]
```diff
@@ -165,7 +166,15 @@ function successfulTypeScriptInstance(
       loader.context
     );
 
-    loader._module.errors.push(...errors);
+    /**
+     * Since webpack 5, the `errors` property is deprecated,
+     * so we can check if some methods for reporting errors exist.
+     */
+    if (!!module.addError) {
```
Comment (johnnyreilly) [inline src/instances.ts:173]: Do we need the `!!` here?

```suggestion
    if (module.addError) {
```

File: src/instances.ts [hunk 3]
```diff
@@ -431,7 +441,15 @@ export function reportTranspileErrors(
       { file: instance.configFilePath || 'tsconfig.json' },
       loader.context
     );
-    loader._module.errors.push(...solutionErrors, ...errors);
+    /**
+     * Since webpack 5, the `errors` property is deprecated,
+     * so we can check if some methods for reporting errors exist.
+     */
+    if (!!module.addError) {
```
Comment (johnnyreilly) [inline src/instances.ts:448]: Do we need the `!!` here?

```suggestion
    if (module.addError) {
```

**Comment by johnnyreilly** [comment]
Looks good!  I've left some comments / suggestions above.  What do you think?

File: src/instances.ts (hunk 3 above)
Comment (sanex3339) [reply src/instances.ts]: Fixed in https://github.com/TypeStrong/ts-loader/pull/1195/commits/f0a81318d255c18f11027e043d3bd56b13460bad

File: src/instances.ts (hunk 2 above)
Comment (sanex3339) [reply src/instances.ts]: Fixed in https://github.com/TypeStrong/ts-loader/pull/1195/commits/f0a81318d255c18f11027e043d3bd56b13460bad

File: src/index.ts (hunk 1 above)
Comment (sanex3339) [reply src/index.ts]: Fixed in https://github.com/TypeStrong/ts-loader/pull/1195/commits/f0a81318d255c18f11027e043d3bd56b13460bad

**Comment by sanex3339** [comment]
> Looks good!  I've left some comments / suggestions above. (Essentially the same point three times :smile: )  What do you think?

Fixed

**Comment by johnnyreilly** [comment]
Could you update the version in the `package.json` to 8.0.5 and add an entry to the `package.json` please? If the tests pass we're likely safe to ship.

**Comment by sanex3339** [comment]
> Could you update the version in the `package.json` to 8.0.5 and add an entry to the `package.json` please? If the tests pass we're likely safe to ship.

https://github.com/TypeStrong/ts-loader/pull/1195/commits/57d3fe946692a9904ac19e317904eab8c9d24521

**Pull request merged**